packages/dds/sequence/src/test/intervalRevertibles.fuzz.spec.ts

```diff
@@ -0,0 +1,376 @@
+		return { type: "removeRange", ...exclusiveRange(state) };
+	}
+
+	async function addInterval(state: ClientOpState): Promise<AddInterval> {
+		return {
+			type: "addInterval",
+			...inclusiveRange(state),
+			collectionName: state.random.pick(options.intervalCollectionNamePool),
+			id: state.random.uuid4(),
+			// added to get rid of compiler errors - assume we
+			// don't want to test both features here
+			stickiness: 0,
+		};
+	}
+
+	async function deleteInterval(state: ClientOpState): Promise<DeleteInterval> {
+		return {
+			type: "deleteInterval",
+			...interval(state),
+		};
+	}
+
+	async function changeInterval(state: ClientOpState): Promise<ChangeInterval> {
+		const { start, end } = inclusiveRange(state);
+		return {
+			type: "changeInterval",
+			start: state.random.integer(0, 5) === 5 ? undefined : start,
+			end: state.random.integer(0, 5) === 5 ? undefined : end,
+			...interval(state),
+		};
+	}
+
+	async function changeProperties(state: ClientOpState): Promise<ChangeProperties> {
+		return {
+			type: "changeProperties",
+			...interval(state),
+			properties: propertySet(state),
+		};
+	}
+
+	async function revertSharedStringRevertibles(
+		state: ClientOpState,
+	): Promise<RevertSharedStringRevertibles> {
+		return {
+			type: "revertSharedStringRevertibles",
+			...interval(state),
+		};
+	}
+
+	const hasAnInterval = ({ channel }: ClientOpState): boolean =>
+		Array.from(channel.getIntervalCollectionLabels()).some((label) => {
+			const collection = channel.getIntervalCollection(label);
+			return isNonEmpty(collection);
+		});
+
+	const lengthSatisfies =
+		(criteria: (length: number) => boolean): AcceptanceCondition<ClientOpState> =>
+		({ channel }) =>
+			criteria(channel.getLength());
+	const hasNonzeroLength = lengthSatisfies((length) => length > 0);
+	const isShorterThanMaxLength = lengthSatisfies((length) => length < options.maxStringLength);
+
+	const hasNotTooManyIntervals: AcceptanceCondition<ClientOpState> = ({ channel }) => {
+		let intervalCount = 0;
+		for (const label of channel.getIntervalCollectionLabels()) {
+			for (const _ of channel.getIntervalCollection(label)) {
+				intervalCount++;
+				if (intervalCount >= options.maxIntervals) {
+					return false;
+				}
+			}
+		}
+		return true;
+	};
+
+	const all =
+		<T>(...clauses: AcceptanceCondition<T>[]): AcceptanceCondition<T> =>
+		(t: T) =>
+			clauses.reduce<boolean>((prev, cond) => prev && cond(t), true);
+
+	return createWeightedGenerator<RevertOperation, ClientOpState>([
+		[addText, 2, isShorterThanMaxLength],
+		[removeRange, 1, hasNonzeroLength],
+		// [addInterval, 0, all(hasNotTooManyIntervals, hasNonzeroLength)],
+		[addInterval, 2, all(hasNotTooManyIntervals, hasNonzeroLength)],
+		// [deleteInterval, 2, hasAnInterval],
+		// [changeInterval, 2, all(hasAnInterval, hasNonzeroLength)],
+		[changeProperties, 2, hasAnInterval],
+		// don't know what acceptance condition i should be using
```

Comment:
I think you probably want `hasRevertibles`, which checks that the revertibles stack for the selected sharedstring has content